vendor/src/github.com/docker/swarmkit/manager/controlapi/service.go

```diff
@@ -349,6 +350,11 @@ func (s *Server) UpdateService(ctx context.Context, request *api.UpdateServiceRe
 		if request.Spec != nil && reflect.TypeOf(service.Spec.Mode) != reflect.TypeOf(request.Spec.Mode) {
 			return errModeChangeNotAllowed
 		}
+
+		if request.Spec != nil && service.Spec.Annotations.Name != request.Spec.Annotations.Name {
```

Comment:
The `request.Spec != nil` check is not necessary, and we can also remove the others above. `validateServiceSpec` ensures that it is not `nil`.